### PR TITLE
fix: hidden deleted endpoint

### PR DESCRIPTION
**Pull request by elonniu**
## Description

- Hidden deleted endpoint in delete endpoint list.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: middleware_api/lambda/inference_v2/sagemaker_endpoint_api.py [hunk 1]
```diff
@@ -42,6 +42,10 @@ def list_all_sagemaker_endpoints(event, ctx):
         event['x-auth']['role'] = user_roles
 
     for row in scan_rows:
+        # if status is deleted then skip
+        # todo: will improved this later
+        if 'status' in row and row['status']['S'] == 'deleted':
+            continue
```
Comment (alvindaiyan) [inline middleware_api/lambda/inference_v2/sagemaker_endpoint_api.py:48]: list endpoint should be able to filter the all endpoints not just inService endpoint

Comment (elonniu) [reply middleware_api/lambda/inference_v2/sagemaker_endpoint_api.py:48]: The page also needs to show Creating

File: middleware_api/lambda/inference_v2/sagemaker_endpoint_api.py [hunk 2]
```diff
@@ -42,6 +42,11 @@ def list_all_sagemaker_endpoints(event, ctx):
         event['x-auth']['role'] = user_roles
 
     for row in scan_rows:
+
+        # Compatible with fields used in older versions
+        if 'status' in row and row['status']['S'] == 'deleted':
+            row['endpoint_status']['S'] = 'deleted'
+
```
Comment (alvindaiyan) [inline middleware_api/lambda/inference_v2/sagemaker_endpoint_api.py:49]: suggestion: Use parsed endpoint below

**Review by alvindaiyan — APPROVED** [review]
code style is not consistent, but good to me

**Pull request merged**